front/lib/swr.ts

```diff
@@ -25,8 +25,13 @@ import { DataSourceType } from "@app/types/data_source";
 import { RunRunType } from "@app/types/run";
 import { WorkspaceType } from "@app/types/user";
 
-export const fetcher = (...args: Parameters<typeof fetch>) =>
-  fetch(...args).then((res) => res.json());
+import { HTTPError } from "./error";
+
+export const fetcher = async (...args: Parameters<typeof fetch>) =>
+  fetch(...args).then(async (res) => {
```

Comment:
As per IRL let's not use a new type for now and subsequently introduce a type that takes into account the fact that the errors in most cases should be apiErrors from front API